### ISSUE TITLE
[REQ] Ability to synchronise req-cnts between wiki and code

### ISSUE THREAD

**Issue by mhatzl** [mhatzl/mantra#1]
mantra must have a command to automatically update ref-cnts in the wiki.
Updating these cnts manually would not be feasible.

Preferably a GitHub action should be available that updates the wiki on every push to the branch the action is set to.
However, it should also be possible to synchronise ref-cnts for local projects and wikis.

**Comment by mhatzl** [comment]
Wiki pages are simple markdown files in GitHub, so there are no property fields counts can be attached to.
As a result, some syntax rules must be adhered to by users for mantra to correctly find and update ref-cnts in the wiki.

The best way I could come up with so far is a bullet list with `**References:**` above it, which is directly below the heading of the requirement.

**Example:**

```
# my_req: Some Title

**References:**

- in branch main: 3
- in branch stable: 1
```

The branches should also have links pointing to the branch in the repository.

**Comment by mhatzl** [comment]
Requirements will be traced using a *requirements ID* set in the wiki section.
To allow hierarchical requirement structures, the ID of sub-requirements must follow the form `parent_reg_id.sub_req_id`.
As consequence, a `.` in an ID is only allowed if requirement IDs exist for all *parent* ID-parts.

**Comment by mhatzl** [comment]
Syntax for references on the code side will be `[req:your_req_id]`.
This should be distinctive enough from common expressions in most programming languages.

**Comment by mhatzl** [comment]
This issue was partially implemented in [v0.1.0](https://github.com/mhatzl/mantra/commit/80568bb23e2c3a5243392f74a267af6a54194deb).

The GitHub action will take more time, and is moved to the separate issue #11
